packages/admin/cms-admin/src/contentScope/ContentScopeSelect.tsx

```diff
@@ -0,0 +1,177 @@
+import { AppHeaderDropdown, ClearInputAdornment } from "@comet/admin";
+import { Domain, Search } from "@comet/admin-icons";
+import { Box, Divider, InputAdornment, InputBase, List, ListItemButton, ListItemText, ListSubheader } from "@mui/material";
+import { capitalCase } from "change-case";
+import React from "react";
+import { useIntl } from "react-intl";
+
+import { ContentScopeInterface } from "./Provider";
+
+type Option<Value extends ContentScopeInterface = ContentScopeInterface> = {
+    [Key in keyof Value]: { label?: string; value: Value[Key] };
+};
+
+interface Props<Value extends ContentScopeInterface> {
+    value: Value;
+    onChange: (value: Value) => void;
+    options: Array<Option<Value>>;
+    searchable?: boolean;
+    groupBy?: keyof Value;
+    icon?: React.ReactNode;
+    renderOption?: (option: Option<Value>) => React.ReactNode;
+    renderSelectedOption?: (option: Option<Value>) => React.ReactNode;
+}
+
+export function ContentScopeSelect<Value extends ContentScopeInterface = ContentScopeInterface>({
+    value,
+    onChange,
+    options,
+    searchable,
+    groupBy,
+    icon = <Domain />,
+    renderOption,
+    renderSelectedOption,
+}: Props<Value>) {
+    const intl = useIntl();
+    const [searchValue, setSearchValue] = React.useState<string>("");
+
+    const hasMultipleDimensions = Object.keys(value).length > 1;
+
+    let filteredOptions: Array<Option<Value>>;
+
+    if (searchable) {
+        filteredOptions = options.filter((option) => {
+            return Object.values(option).some(({ label, value }) => {
+                return value.toLowerCase().includes(searchValue.toLowerCase()) || label?.toLowerCase().includes(searchValue.toLowerCase());
+            });
+        });
+    } else {
+        filteredOptions = options;
+    }
```

Comment:
Could be slightly simplified. 

```suggestion
    let filteredOptions: Array<Option<Value>> = options;

    if (searchable) {
        filteredOptions = options.filter((option) => {
            return Object.values(option).some(({ label, value }) => {
                return value.toLowerCase().includes(searchValue.toLowerCase()) || label?.toLowerCase().includes(searchValue.toLowerCase());
            });
        });
    }
```

Reply:
2fc690077dc396b3eef2d764666756283d18d50f